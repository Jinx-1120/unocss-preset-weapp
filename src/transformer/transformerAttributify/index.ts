import type { Options } from 'unplugin-attributify-to-class/types'
import transformer from './transformer'
import { presetWeappAttributify as preset } from './autocomplete'

export function extractorAttributify(options: Options = {}) {
  return {
    transformerAttributify: () => transformer(options),
    presetWeappAttributify: () => preset(options),
  }
}

export const transformerAttributify = transformer
