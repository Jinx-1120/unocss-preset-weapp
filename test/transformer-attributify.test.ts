import { createGenerator } from '@unocss/core'
import { describe, expect, test } from 'vitest'
import MagicString from 'magic-string'

import { transformerAttributify } from '../src/transformer'
import { presetWeapp } from '../src'

const originalCode = `
<template>
<view
  text="sm green"
  p="y-2 x-4"
  m="4"
>
xx
</view>

<view flex="~ col wrap" class="m4">
  xxx
</view>


<view even:bg-red odd:bg-green>
  1
</view>

<view first:bg-red last:bg-green bg-blue mb-1>
  1
</view>
</template>
`

describe('transformerAttributify', () => {
  const uno = createGenerator({
    presets: [
      presetWeapp(),
    ],
  })

  test('transform', async () => {
    const code = new MagicString(originalCode)
    await transformerAttributify().transform(code, 'app.vue', { uno, tokens: new Set() } as any)
    await expect(code.toString()).toMatchFileSnapshot('./assets/output/transformer-attributify.vue')
  })
})
