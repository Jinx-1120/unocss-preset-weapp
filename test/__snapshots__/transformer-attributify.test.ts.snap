// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transformerAttributify > transform 1`] = `
"
<template>
<view
  text=\\"sm green\\"
  p=\\"y-2 x-4\\"
  m=\\"4\\"
 class=\\"text-sm text-green p-y-2 p-x-4 m-4\\">
xx
</view>

<view flex=\\"~ col wrap\\" class=\\"m4 flex flex-col flex-wrap\\">
  xxx
</view>


<view even:bg-red odd:bg-green class=\\"even_cl_bg-red odd_cl_bg-green\\">
  1
</view>

<view first:bg-red last:bg-green bg-blue mb-1 class=\\"first_cl_bg-red last_cl_bg-green bg-blue mb-1\\">
  1
</view>
</template>
"
`;
